app/models/ext_management_system.rb

```diff
@@ -260,6 +264,78 @@ def default_endpoint
     default || endpoints.build(:role => "default")
   end
 
+  # TODO: takes multiple connection data
+  # endpoints, authentications, and works out what's what.
+  def connections=(options)
+    options.each do |option|
+      endpoint = endpoint_builder(option[:endpoint])
+      authentication_builder(option[:authentication])
+        # connection_by_role=(option)
+    end
+  end
+
+  def connections
```

Comment:
I just had a chat with @juliancheal and we think `connections` is not a really good wording in this case.
I thought of it returning actual connections to something, but what it does is returning a configuration for an endpoint plus authentication for that endpoint.
I'm not sure how to name it better, though.

I like the domain context of an endpoint in this case. It conveys the meaning of being a configuration (like host and port) for an ExtManagementSystem.
And, while `connections` throws in authentication for it, I rather think that the authentication is actually a part of the endpoint. (although one could imagine authentications shared by endpoints) 

```ruby
# Amazon::CloudManager
def connect
  raw_connect(host: endpoints.default.hostname, username: endpoints.default.username)
end
```

or, if authentication is separated out

```ruby
# Amazon::CloudManager
def connect
  raw_connect(host: endpoint_configurations.default.hostname, username: endpoint_configurations.default.authentication.username)
end
```


@blomquisg @Fryguy @chessbyte thoughts?